builder/steps/gen-dockerfile/contents/data/Dockerfile.txt

```diff
@@ -1,14 +1,52 @@
+<% if (config.buildCommand) { %>
+FROM <%- baseImage %> as build_step
+COPY . /app/
+<% if (config.canInstallDeps) { %>
+ENV NODE_ENV development
+  <% if (config.useYarn) { %>
+RUN yarn install || \
+  ((if [ -f yarn-error.log ]; then \
+      cat yarn-error.log; \
+    fi) && false)
+  <% } else { %>
+# You have to specify "--unsafe-perm" with npm install
+# when running as root.  Failing to do this can cause
+# install to appear to succeed even if a preinstall
+# script fails, and may have other adverse consequences
+# as well.
+# This command will also cat the npm-debug.log file after the
+# build, if it exists.
+RUN npm install --unsafe-perm || \
+  ((if [ -f npm-debug.log ]; then \
+      cat npm-debug.log; \
+    fi) && false)
+  <% } %>
+ENV NODE_ENV production
+<% } %>
+RUN <%- config.buildCommand %>
```

Comment:
`RUN cp /etc/passwd /pwned` would get the /etc/passwd from inside the container, whereas `COPY /etc/passwd /pwned` would let the user reach out to the Argo build VM, and copy arbitrary stuff into the container for later perusal.  I don't know what specifically is sensitive inside the build VM, but I haven't seen a document that says "0% guaranteed, it's fine" either.

If nothing else, we don't want users to realize there is a "feature" of "Can inject arbitrary Dockerfile directives" and then start using and depending on that "feature"